File: study/lua/src/app/data/GuildTaskUnitData.lua

local BaseData = require("app.data.BaseData")
local GuildTaskUnitData = class("GuildTaskUnitData", BaseData)

local schema = {}
schema["id"] 			= {"number", 0}  --id
schema["people"] 		= {"number", 0}  --完成任务人数
schema["config"] 		= {"table", nil} --配置
schema["exp"] 		= {"number", 0}  --经验值

GuildTaskUnitData.schema = schema

function GuildTaskUnitData:ctor(properties)
	GuildTaskUnitData.super.ctor(self, properties)
end

function GuildTaskUnitData:clear()
end

function GuildTaskUnitData:reset()
end

function GuildTaskUnitData:initData(config)
    self:setConfig(config)
    self:setId(config.type)
    self:setPeople(0)
    self:setExp(0)

end


return GuildTaskUnitData

File: study/ts/script/data/GuildTaskUnitData.ts
import { BaseData } from "./BaseData";

export interface GuildTaskUnitData {
    getId(): number
    setId(value: number): void
    getLastId(): number
    getPeople(): number
    setPeople(value: number): void
    getLastPeople(): number
    getConfig(): Object
    setConfig(value: Object): void
    getLastConfig(): Object
    getExp(): number
    setExp(value: number): void
    getLastExp(): number
}
let schema = {};
schema['id'] = [
    'number',
    0
];
schema['people'] = [
    'number',
    0
];
schema['config'] = [
    'object',
    null
];
schema['exp'] = [
    'number',
    0
];
export class GuildTaskUnitData extends BaseData {
    public static schema = schema;

    constructor(properties?) {
        super(properties);
    }
    public clear() {
    }
    public reset() {
    }
    public initData(config) {
        this.setConfig(config);
        this.setId(config.type);
        this.setPeople(0);
        this.setExp(0);
    }
}
